src/bytes_mut.rs

```diff
@@ -468,17 +468,27 @@ impl BytesMut {
     /// assert_eq!(&buf[..], &[0x1, 0x1, 0x3, 0x3]);
     /// ```
     pub fn resize(&mut self, new_len: usize, value: u8) {
-        let len = self.len();
-        if new_len > len {
-            let additional = new_len - len;
-            self.reserve(additional);
-            unsafe {
-                let dst = self.chunk_mut().as_mut_ptr();
-                ptr::write_bytes(dst, value, additional);
-                self.set_len(new_len);
-            }
+        let additional = if let Some(additional) = new_len.checked_sub(self.len()) {
+            additional
         } else {
             self.truncate(new_len);
+            return;
+        };
+
+        if additional == 0 {
+            return;
+        }
+
+        self.reserve(additional);
+        let dst = self.spare_capacity_mut().as_mut_ptr();
+        unsafe {
+            // SAFETY: `spare_capacity_mut` returns a valid, properly aligned pointer and we've
+            // reserved enough space to write `additional` bytes.
+            ptr::write_bytes(dst, value, additional);
+
+            // SAFETY: There are at least `new_len` initialized bytes in the buffer so no
+            // uninitialized bytes are being exposed.
+            self.set_len(new_len);
         }
```

Comment:
It's great that you are adding safety comments! I think it is also good to take the opportunity to reduce the scope of these unsafe blocks.
```suggestion
        // SAFETY: `spare_capacity_mut` returns a valid, properly aligned pointer and we've
        // reserved enough space to write `additional` bytes.
        unsafe { ptr::write_bytes(dst, value, additional) };

        // SAFETY: There are at least `new_len` initialized bytes in the buffer so no
        // uninitialized bytes are being exposed.
        unsafe { self.set_len(new_len) };
```

Reply:
Perfect. I also prefer to isolate them as much as possible but I wasn't sure of the conventions here. Thanks @Darksonn!